pokemongo_bot/cell_workers/transfer_pokemon.py

```diff
@@ -33,41 +37,69 @@ def work(self):
                     for best_pokemon_id in best_pokemon_ids:
-                        for pokemon in all_pokemons:
+                        for pokemon in group:
                             if best_pokemon_id == pokemon['pokemon_data']['id']:
-                                all_pokemons.remove(pokemon)
-                                best_pokemons.append(pokemon)
-
-                    transfer_pokemons = [pokemon for pokemon in all_pokemons
-                                         if self.should_release_pokemon(pokemon_name,
-                                                                        pokemon['cp'],
-                                                                        pokemon['iv'],
-                                                                        True)]
+                                group.remove(pokemon)
+                                best_pokemon.append(pokemon)
 
-                    if transfer_pokemons:
-                        logger.log("Keep {} best {}, based on {}".format(len(best_pokemons),
+                    if len(best_pokemon) > 0:
+                        logger.log("Keep {} best {}, based on {}".format(len(best_pokemon),
                                                                          pokemon_name,
                                                                          order_criteria), "green")
-                        for best_pokemon in best_pokemons:
+                        for best_pokemon in best_pokemon:
                             logger.log("{} [CP {}] [Potential {}]".format(pokemon_name,
                                                                           best_pokemon['cp'],
                                                                           best_pokemon['iv']), 'green')
 
-                        logger.log("Transferring {} pokemon".format(len(transfer_pokemons)), "green")
+                high_pokemon = []
+                for pokemon in group:
+                    if self.should_release_pokemon(pokemon_name, pokemon['cp'], pokemon['iv']):
+                        group.remove(pokemon)
+                        high_pokemon.append(pokemon)
+                if len(high_pokemon) > 0:
+                    logger.log("Keep {} {}, based on cp/iv criteria".format(len(high_pokemon),
+                                                                            pokemon_name), "green")
+                    for high_pokemon in high_pokemon:
+                        logger.log("{} [CP {}] [Potential {}]".format(pokemon_name,
+                                                                      high_pokemon['cp'],
+                                                                      high_pokemon['iv']), 'green')
+
+                if keep_for_evo and len(group) > 0:
+
+                    if candies == {}:
+                        logger.log("Api call for candies failed, try again")
+                        return
+                    candy = candies[pokemon_id]
```

Comment:
Pokemon with 2 evolutions always use candy of the first kind of pokemon.
We have two solutions use the ['Next Evolution Requirements']['Name'] to order candies array (as my PR #2244 ). Or add the Evolution candy family in the JSON used for pokemon_list variable